### ISSUE TITLE
Branch topography is displayed incorrectly, some merge topography is missing

### ISSUE THREAD

**Issue by Road-hog123** [phil294/GitLG#81]
The expected topography:
```
| * 8881589 Remove unused 'order' property from layout
| * 6e3211c (tag: v1.0.0) Release prep
|/
* 12215d0 Initial Commit
* 0d2f992 Setup Repository
```
Instead the branch diverges before the parent commit:
![image](https://github.com/phil294/git-log--graph/assets/12716638/620eb4ad-0562-486c-b6b6-0d1a85034ef8)

Merges have the correct topography in `--topo-order`, albeit uncomfortably close to the previous commit:
```
* 7bf4247b095 Depsgraph: add a DEG_graph_build_from_collection convenience function
*   8c16b612ca8 Merge branch 'blender-v4.1-release'
|\
* \   5af49874565 Merge branch 'blender-v4.1-release'
|\ \
* | | 1b514659cae Cleanup: minor changes to temp directory API
* | |   b0daa792d3b Merge branch 'blender-v4.1-release'
|\ \ \
* | | | bf17fc8d797 Fix: GPU: Ensures length of curves GPUIndexBuf to be multiple of 4
* | | | 3d136d0d003 BLI: Add support for non-square matrix multiplication.
* | | |   62bb346af91 Merge branch 'blender-v4.1-release'
|\ \ \ \
* \ \ \ \   fdc9bcf523f Merge branch 'blender-v4.1-release'
|\ \ \ \ \
* \ \ \ \ \   10b5fe88977 Merge branch 'blender-v4.1-release'
|\ \ \ \ \ \
* \ \ \ \ \ \   1756ccf8098 Merge branch 'blender-v4.1-release'
|\ \ \ \ \ \ \
* \ \ \ \ \ \ \   434759d3339 Merge branch 'blender-v4.1-release'
|\ \ \ \ \ \ \ \
* | | | | | | | | 6db0bee6854 Cleanup: Add utility for volume grid nodes search items
```
![image](https://github.com/phil294/git-log--graph/assets/12716638/bb1ccdac-50eb-452a-957d-d7cf97a32873)
Setting `curve-radius` to `0` merely highlights the weird curviness and disjoints:
![image](https://github.com/phil294/git-log--graph/assets/12716638/fd9f61d0-ee79-48a7-9b4d-e8ad9008c85f)

When displayed in date order these branches have a bunch of crossovers:
```
* | 7bf4247b095 Depsgraph: add a DEG_graph_build_from_collection convenience function
* | 8c16b612ca8 Merge branch 'blender-v4.1-release'
|\|
| * 7e9a36c1fa3 Cleanup: suppress undefined variable use warnings
* | 5af49874565 Merge branch 'blender-v4.1-release'
|\|
| * 51126fab331 BLI_tempfile: ensure the temporary directory is absolute
* | 1b514659cae Cleanup: minor changes to temp directory API
* | b0daa792d3b Merge branch 'blender-v4.1-release'
|\|
| * 8f68dfe0c24 Fix #119000: File handler extension check should be case insensitive
* | bf17fc8d797 Fix: GPU: Ensures length of curves GPUIndexBuf to be multiple of 4
* | 3d136d0d003 BLI: Add support for non-square matrix multiplication.
* | 62bb346af91 Merge branch 'blender-v4.1-release'
|\|
| * 5f70bd0e46b Fix #116435: Rotate Vector node on a link connects wrong sockets
* | fdc9bcf523f Merge branch 'blender-v4.1-release'
|\|
| * cce2b6671ec Fix #114110: background dots clipping in node editor
* | 10b5fe88977 Merge branch 'blender-v4.1-release'
|\|
| * e72bf54791a Fix #118926: missing transform relation when using Object/Collection Info node
* | 1756ccf8098 Merge branch 'blender-v4.1-release'
|\|
| * 65083472ecd Fix: issue in previous commit
* | 434759d3339 Merge branch 'blender-v4.1-release'
|\|
| * 173c41c1c07 Fix #114946: Mesh.clear_geometry should not clear required built-in attributes
* | 6db0bee6854 Cleanup: Add utility for volume grid nodes search items
```
Which are entirely missing (and the merge commits are no longer shaded grey):
![image](https://github.com/phil294/git-log--graph/assets/12716638/da8b9eca-8072-46f7-9321-c3e177d5749e)

**Comment by phil294** [comment]
thanks for the report. this looks a bit difficult to debug as I think this isn't reproducible on just any kind of merge commit. 

> Instead the branch diverges before the parent commit:

Can you provide the repository you've demoed this with, and the exakt git cli command you have entered to get the graph output? So it is easier for me to compare both

**Comment by phil294** [comment]
ah wait, it *does* seem to be erroneous for every kind of 

```
|/
*
```
branch birth commit, the line starts one commit too early. This doesn't seem critical to me as you normally don't really look at it and the more important junctions, the merge commits, seem to be rendered correctly. Should be fixed either way.

about the rest of your message, I'm not sure yet

**Comment by J-Sorenson** [comment]
I would say that having the "branch-from" line in the correct location is rather critical, as it is giving wrong information.  I had to tell the other users to make note of it, as the chart suggests a previous "fix" commit is *not* included in their feature branch when it actually is.  Preferably, we'd have the line's branching from/to the commit circle itself, but I understand your line-connect method is quite a bit easier to implement graphically.  I'm not a javascript developer, so I don't know how difficult it would be to adjust the position of the join-location so that it is directly above the "branch" and directly below the "merge."
That said: we can mentally adjust in the meantime.  I'm just glad this git-graph replacement initiative exists!

**Comment by phil294** [comment]
Should be fixed with v0.1.20 later today, both the wrong birth branch positioning and missing merge commits. Please give it a try.

> Preferably, we'd have the line's branching from/to the commit circle itself

in both directions too.

> I don't know how difficult it would be to adjust the position of the join-location so that it is directly above the "branch" and directly below the "merge."

the thing is, each row in the main view comes from one or more respective rows of `git log --graph --oneline --all` output and has its own thin graph visualization lines section for performance reasons and they can't expand into above or below rows. But I did add special logic so that a small joining line is retroactively added to the above commit, so all in all not really difficult no
